### PR TITLE
Updated add_block() and magic_bytes

### PR DESCRIPTION
**Pull request by jldistefano**
NOTE: This issue was primarily completed by @JoshuaRM , and my contribution to this issue only updates documentation and test functions to project standards.

### CODE REVIEW

File: data_structures/blockchain.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import os.path
 from struct import pack
+
+magic_bytes = pack('L', 3652501241)
```
Comment (kstola2) [inline data_structures/blockchain.py]: import `int_to_bytes` from block and use that instead here

File: data_structures/blockchain.py [hunk 2]
```diff
@@ -41,4 +44,16 @@ def extract(filename, index, num_bytes, offset=0):
       # Moves file pointer to correct position at index + offset
       file.seek(index, offset)
       # Reads num_bytes after file pointer
-      return file.read(num_bytes)
\ No newline at end of file
+      return file.read(num_bytes)
+    
+def add_block(self, block):
```
Comment (kstola2) [inline data_structures/blockchain.py]: This is outside of the class so the tests aren't detecting an `add_block` function inside the `Blockchain` class.

File: data_structures/blockchain.test.py [hunk 3]
```diff
@@ -57,5 +57,42 @@ def test_extract(self):
...
+			# Extracts block from blockchain
+			actual = extract(self.bc.blockfile, 0, size)
+			# Confirms that extracted block is identical to actual block
+			self.assertEqual(expected, actual)
+		
+	def test_add_block2(self):
+		target = 10**72     
+		
+		# Creates 4 blocks and add to file
+		b1 = mine(hash_SHA("Root.encode"()), "Block1".encode(), target)
```
Comment (kstola2) [inline data_structures/blockchain.test.py]: `"Root.encode"` needs to be fixed syntactically

File: data_structures/blockchain.test.py [hunk 4]
```diff
@@ -57,5 +57,42 @@ def test_extract(self):
...
+		self.bc.add_block(b1)
+		b2 = mine(hash_SHA"Block1".encode()), "Block2".encode(), target)
```
Comment (kstola2) [inline data_structures/blockchain.test.py]: Uhhh I think there's a parentheses missing in `hash_SHA`

File: data_structures/blockchain.test.py [hunk 5]
```diff
@@ -57,5 +57,42 @@ def test_extract(self):
...
+		self.bc.add_block(b2)
+		b3 = mine(hash_SHA"Block2".encode()), "Block3".encode(), target)
+		self.bc.add_block(b3)
+		b4 = mine(hash_SHA"Block3".encode()), "Block4".encode(), target)
+		self.bc.add_block(b4)
+		# Expected value should be concatonation of all blocks
+		expected = b1 + b2 + b3 + b4
+		#Get size of block
+		size = get_size_bytes(expected)
```
Comment (kstola2) [inline data_structures/blockchain.test.py]: `get_size_bytes` returns a byte string so this needs to be converted to an int first. You can import `bytes_to_int` from block.

**Comment by jldistefano** [comment]
Requested changes made

**Pull request closed without merging**